iwyu.cc

```diff
@@ -3933,6 +3933,16 @@ class IwyuAstConsumer
     return Base::VisitTypedefType(type);
   }
 
+  bool VisitEnumDecl(clang::EnumDecl* decl) {
+    clang::QualType integer_type = decl->getIntegerType();
+    if (!integer_type.isNull()) {
```

Comment:
You can say `if (const clang::Type* type = integer_type.getTypePtrOrNull())` to get both of these conditions in a single expression.